### ISSUE TITLE
Missing dependencies within useEffect hook

### ISSUE THREAD

**Issue by patrykkarny** [underscopeio/react-native-navigation-hooks#4]
hey!

I discovered that if we want to use ie. `useNavigationButtonPress` with a `handler` that closing some variables from the outside scope, it doesn't work. It is because of `useEffect` missing `handler` as a dependency. For example:

```js
useNavigationButtonPress(
  () => deleteSomething(somethingId), // it will work only on first render
  'componentId',
  'buttonId'
)
```
but if we add handler as a dependency to `useEffect`, then on rerender it will use newly created function with refreshed `somethingId`.

```js
function useNavigationButtonPress(handler, componentId, buttonId) {
  useEffect(() => {
  // useNavigationButtonPress body
  }, [handler])
}
```

**Comment by jpgarcia** [comment]
You're right! actually a couple of days ago I noticed the same issue. I'll try to perform the change when I have free time. 
What we add a new parameter to pass a dependency array instead of a handler?  

Something like:
```js
useNavigationButtonPress(
  () => deleteSomething(somethingId), // it will work only on first render
  'componentId',
  'buttonId',
  [somethingId]
)
```
Does it makes sense?

**Comment by patrykkarny** [comment]
hmm I'm not sure how you would like to do it but I think it won't work because you end up passing handler as a dependency of `useEffect` anyway

**Comment by alayek** [comment]
Just came across this repo, marvellous work! Would surely help someone trying to uptake wix/react-native-navigation with hooks.

What if we used a `ref` to always get latest `handler`?

#### Original code

```
function useNavigationButtonPress(handler, componentId, buttonId) {
  useEffect(() => {
    const subscription = Navigation.events().registerNavigationButtonPressedListener(event => {
      const equalComponentId = event.componentId === componentId
      const equalButtonId = event.buttonId === buttonId

      if (equalComponentId && equalButtonId) {
        handler(event)
      } else if (!buttonId && equalComponentId) {
        handler(event)
      } else if (!componentId && equalButtonId) {
        handler(event)
      } else if (!componentId && !buttonId) {
        handler(event)
      }
    })

    return () => subscription.remove()
  }, [])
}
```

#### proposal

```
function useNavigationButtonPress(handler, componentId, buttonId) {
  const handlerRef = React.useRef(handler); // initialize with first version of handler passed

  React.useEffect(() => {
      handlerRef.current = handler; // update
  }, [handler]);

  React.useEffect(() => {
    const subscription = Navigation.events().registerNavigationButtonPressedListener(event => {
      const equalComponentId = event.componentId === componentId
      const equalButtonId = event.buttonId === buttonId

      if (equalComponentId && equalButtonId) {
        handlerRef.current(event)
      } else if (!buttonId && equalComponentId) {
        handlerRef.current(event)
      } else if (!componentId && equalButtonId) {
        handlerRef.current(event)
      } else if (!componentId && !buttonId) {
        handlerRef.current(event)
      }
    })

    return () => subscription.remove()
  }, [])
}
```

I'm new to hooks, so apologies and correct me if I'm wrong.

**Comment by jpgarcia** [comment]
Hey Guys! 

I just released a new version [3.0.0](https://github.com/underscopeio/react-native-navigation-hooks/releases/tag/v3.0.0) which includes this improvement. Also I've added a note to the [readme](https://github.com/underscopeio/react-native-navigation-hooks#memoize-your-handlers) file suggesting the use of the `useCallback` hook.

Thank you for all your suggestions! and let me know if worked
